ts/node/database_utility.ts

```diff
@@ -0,0 +1,183 @@
+import { difference, omit, pick } from 'lodash';
+import { ConversationAttributes } from '../models/conversationAttributes';
+
+export function objectToJSON(data: Record<any, any>) {
+  const str = JSON.stringify(data);
+  return str;
+}
+export function jsonToObject(json: string): Record<string, any> {
+  return JSON.parse(json);
+}
+
+function jsonToArray(json: string): Array<string> {
+  try {
+    return JSON.parse(json);
+  } catch (e) {
+    console.warn('jsontoarray failed:', e.message);
+    return [];
+  }
+}
+
+export function arrayStrToJson(arr: Array<string>): string {
+  return JSON.stringify(arr);
+}
+
+export function toSqliteBoolean(val: boolean): number {
+  return val ? 1 : 0;
+}
+
+// this is used to make sure when storing something in the database you remember to add the wrapping for it in formatRowOfConversation
+const allowedKeysFormatRowOfConversation = [
+  'groupAdmins',
+  'members',
+  'zombies',
+  'isTrustedForAttachmentDownload',
+  'isPinned',
+  'isApproved',
+  'didApproveMe',
+  'is_medium_group',
+  'mentionedUs',
+  'isKickedFromGroup',
+  'left',
+  'lastMessage',
+  'lastMessageStatus',
+  'triggerNotificationsFor',
+  'unreadCount',
+  'lastJoinedTimestamp',
+  'subscriberCount',
+  'expireTimer',
+  'active_at',
+  'id',
+  'type',
+  'avatarPointer',
+  'avatarHash',
+  'nickname',
+  'profileKey',
+  'avatarInProfile',
+  'displayNameInProfile',
+];
+
+export function formatRowOfConversation(row?: Record<string, any>): ConversationAttributes | null {
+  if (!row) {
+    return null;
+  }
+
+  const foundInRowButNotInAllowed = difference(
+    Object.keys(row),
+    allowedKeysFormatRowOfConversation
+  );
+
+  if (foundInRowButNotInAllowed?.length) {
+    console.warn('formatRowOfConversation: foundInRowButNotInAllowed: ', foundInRowButNotInAllowed);
+
+    throw new Error(
+      `formatRowOfConversation: an invalid key was given in the record: ${foundInRowButNotInAllowed[0]}`
+    );
+  }
+
+  const convo: ConversationAttributes = omit(row, 'json') as ConversationAttributes;
+
+  convo.groupAdmins =
+    convo.groupAdmins?.length && row.groupAdmins.length > 5 ? jsonToArray(row.groupAdmins) : [];
```

Comment:
damn, that should be a 0!

Reply:
of no that's all good actually. This is just to avoid trying to decode "[]" as json. If the length of the stored string is less than 5, just return [] rather than parsing it.

`5` is completely arbitrary though. Just thought that our session ids are 64 chars+ and "[]" makes no sense to be parsed. So anything between those two was fine. I also wasn't sure if JSON.stringify returns "[ ]"  or just "[]" and I actually didn't care. 5 seems fine :D
Will add a comment

Reply:
👍